activerecord/lib/active_record/railtie.rb

```diff
@@ -339,5 +339,44 @@ class Railtie < Rails::Railtie # :nodoc:
         end
       end
     end
+
+    initializer "active_record.query_log_tags_config" do |app|
+      config.after_initialize do
+        if app.config.active_record.query_log_tags_enabled
+
+          ActiveRecord::QueryLogs.taggings.merge! \
+            socket:   -> { ActiveRecord::Base.connection_db_config.socket },
+            db_host:  -> { ActiveRecord::Base.connection_db_config.host },
+            database: -> { ActiveRecord::Base.connection_db_config.database }
+
+          # TODO: find an alternative location for these?
+          name = Rails.application.class.name.split("::").first
+          ActiveRecord::QueryLogs.set_context(application_name: name)
+          ActiveRecord::QueryLogs.backtrace_cleaner = Rails.backtrace_cleaner
+          ActiveRecord::QueryLogs.taggings.merge! \
+            application:  -> { context[:application_name] },
```

Comment:
I'm wondering if we should define this as:

```suggestion
            application:  -> { @application_name ||= Rails.application.class.name.split("::").first },
```